src/lib.rs

```diff
@@ -0,0 +1,60 @@
+//! A first-party support of the `wasm32-wasip1` target for the **Server-Side**
```

Comment:
I might have misunderstood, but I thought you were going for wasip2?

Reply:
Yes, it is actually targeting the Preview 2 of WASI, but I don't see mention of the Tier upgrade in the latest Rust release notes so I am not sure `wasip2` toolchain is ready. I think I should ask to Bytecode Alliance fellows

Reply:
So atm, most people target `wasip1` and use an adapter

Reply:
It probably isn't, that'd be a good question though. I wonder how things would go if you ran cargo component on the server

Reply:
Honestly I haven't tried it out but I would expect it to fail since you need a pretty specific toolchain (that is bundled with the Tier 2 upgrade)

I asked about that there: https://bytecodealliance.zulipchat.com/#narrow/channel/235408-rust-toolchain/topic/Promoting.20wasm32-wasip2.20to.20tier.202/near/478774747

Reply:
Oh Alex confirmed that! 🎉 

https://github.com/rust-lang/rust/pull/126967/

So I can try actually!